app.py

```diff
@@ -209,22 +209,35 @@ def delete(self, meme_id):
                         (args.netid, meme_id))
             return send_error("You haven't voted for meme %s" % meme_id)
 
-    def put(self, meme_id):
+    def put(self, meme_id, vote_type):
         ''' Cast your vote for the requested meme '''
         parser = reqparse.RequestParser()
         parser.add_argument('netid', required=True)
         args = parser.parse_args()
 
+        if not request.json or 'vote_type' not in request.json:
+            vote_type = 'like'
+        else:
+            vote_type = request.json['vote_type']
+
         if not Meme.query.filter_by(id=meme_id).first():
             return unknown_meme_response(meme_id)
 
+        if not vote_type >= 0 and vote_type < len(list(React)):
+            return unknown_react_response(vote_type)
+
         vote = Vote.query.filter_by(
             netid=args.netid, meme_id=meme_id).first()
         if not vote:
-            vote = Vote(netid=args.netid, meme_id=meme_id)
+            vote = Vote(
+                netid=args.netid, meme_id=meme_id, vote_type=React(vote_type))
```

Comment:
Nitpick: Please move the last closing parenthesis on this line to the next line:
 ```
vote = Vote(
    netid=args.netid, meme_id=meme_id, vote_type=React(vote_type)
)
```